app/repos/activity.js

```diff
@@ -56,6 +56,47 @@ const getActivityById = async (id) => {
   }
 }
 
+/**
+ * @typedef Activity
+ * @property {number} activity - e.g. '9'
+ * @property {'sent'|'received'} activityType - e.g. 'sent'
+ * @property {string} pk - e.g. 'ED300100'
+ * @property {string} source - e.g. 'dog'
+ * @property {string} activityDate - e.g. 2024-07-02T00:00:00.000Z
+ * @property {string} targetPk - e.g. 'dog'
+ * @property {string} label - e.g. 'Application pack'
+ */
+/**
+ * @param {string} label
+ * @return {Promise<{
+
+ * }>}
+ */
+const getActivityByLabel = async (label) => {
+  try {
+    const activities = await sequelize.models.activity.findOne({
+      where: { label },
+      include: [{
+        model: sequelize.models.activity_type,
+        as: 'activity_type'
+      },
+      {
+        model: sequelize.models.activity_source,
+        as: 'activity_source'
+      },
+      {
+        model: sequelize.models.activity_event,
+        as: 'activity_event'
+      }]
+    })
+
+    return activities
+  } catch (e) {
+    console.log(`Error retrieving activity for id ${label}:`, e)
```

Comment:
'id' needs to change to 'label' as could be confusing

Reply:
sorted